crates/bevy_pbr/src/ssr/raymarch.wgsl

```diff
@@ -0,0 +1,511 @@
+        max_depth = max(linear_depth, unfiltered_depth);
+        min_depth = min(linear_depth, unfiltered_depth);
+    }
+
+    let bias = 0.000002;
+
+    var res: DistanceWithPenetration;
+    res.distance = max_depth * (1.0 + bias) - ray_depth;
+
+    // distance_fn will be used at the end of the ray march to potentially discard the hit.
+    res.penetration = ray_depth - min_depth;
+
+    if ((*distance_fn).march_behind_surfaces) {
+        res.valid = res.penetration < (*distance_fn).depth_thickness;
+    } else {
+        res.valid = true;
+    }
+
+    return res;
+}
+
+struct DepthRayMarchResult {
+    /// True if the raymarch hit something.
+    hit: bool,
+
+    /// In case of a hit, the normalized distance to it.
+    ///
+    /// In case of a miss, the furthest the ray managed to travel, which could either be
+    /// exceeding the max range, or getting behind a surface further than the depth thickness.
+    ///
+    /// Range: `0..=1` as a lerp factor over `ray_start_cs..=ray_end_cs`.
+    hit_t: f32,
+
+    /// UV correspindong to `hit_t`.
+    hit_uv: vec2<f32>,
+
+    /// The distance that the hit point penetrates into the hit surface.
+    /// Will normally be non-zero due to limited precision of the ray march.
+    ///
+    /// In case of a miss: undefined.
+    hit_penetration: f32,
+
+    /// Ditto, within the range `0..DepthRayMarch::depth_thickness_linear_z`
+    ///
+    /// In case of a miss: undefined.
+    hit_penetration_frac: f32,
+}
+
+struct DepthRayMarch {
+    /// Number of steps to be taken at regular intervals to find an initial intersection.
+    /// Must not be zero.
+    linear_steps: u32,
+
+    /// Exponent to be applied in the linear part of the march.
+    ///
+    /// A value of 1.0 will result in equidistant steps, and higher values will compress
+    /// the earlier steps, and expand the later ones. This might be desirable in order
+    /// to get more detail close to objects in SSR or SSGI.
+    ///
+    /// For optimal performance, this should be a small compile-time unsigned integer,
+    /// such as 1 or 2.
+    linear_march_exponent: f32,
+
+    /// Number of steps in a bisection (binary search) to perform once the linear search
+    /// has found an intersection. Helps narrow down the hit, increasing the chance of
+    /// the secant method finding an accurate hit point.
+    ///
+    /// Useful when sampling color, e.g. SSR or SSGI, but pointless for contact shadows.
+    bisection_steps: u32,
+
+    /// Approximate the root position using the secant method -- by solving for line-line
+    /// intersection between the ray approach rate and the surface gradient.
+    ///
+    /// Useful when sampling color, e.g. SSR or SSGI, but pointless for contact shadows.
+    use_secant: bool,
+
+    /// Jitter to apply to the first step of the linear search; 0..=1 range, mapping
+    /// to the extent of a single linear step in the first phase of the search.
+    /// Use 1.0 if you don't want jitter.
```

Comment:
this is a bit unintuitive, why isnt 0.0 no jitter? i understand this is a straight port, just curious

Reply:
`jitter` is really "length of first step". So no jitter would be 1.